### PR TITLE
[SYCL] Added numa affinity partitioning support to Level Zero

### PR DESCRIPTION
**Pull request by glyons-intel**
Added numa affinity partitioning support to Level Zero.
This aligns Level Zero support with OpenCL support.

Signed-off-by: Gail Lyons <gail.lyons@intel.com>

### CODE REVIEW

File: sycl/test/basic_tests/query.cpp [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+              << devices[i].get_info<info::device::name>() << std::endl;
+    std::cout << "  Platform: "
+              << devices[i].get_platform().get_info<info::platform::name>()
+              << std::endl;
+    std::cout << "  EUs: "
+              << devices[i].get_info<info::device::max_compute_units>()
+              << std::endl;
+    std::vector<device> subdevices;
+    try {
+      std::cout << "  Tiles: ";
```
Comment (smaslov-intel) [inline sycl/test/basic_tests/query.cpp:26]:
```suggestion
      std::cout << "  Sub-devices: ";
```

File: sycl/test/basic_tests/query.cpp [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+          devices[i]
+              .create_sub_devices<
+                  info::partition_property::partition_by_affinity_domain>(
+                  info::partition_affinity_domain::numa);
+      std::cout << subdevices.size() << std::endl;
+    } catch (std::exception) {
+      std::cout << "Error -- cannot create subdevices" << std::endl;
+    }
+    for (int j = 0; j < subdevices.size(); ++j) {
+      std::cout << "  Tile " << j << std::endl;
```
Comment (smaslov-intel) [inline sycl/test/basic_tests/query.cpp:37]:
```suggestion
      std::cout << "  Sub-device " << j << std::endl;
```

File: sycl/test/basic_tests/query.cpp [hunk 3]
```diff
@@ -0,0 +1,43 @@
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -o %t.out
```
Comment (smaslov-intel) [inline sycl/test/basic_tests/query.cpp:1]: I am not sure this test is expected to work for every target, so should it go to llvm-test-suite instead?

File: sycl/test/basic_tests/query.cpp (hunk 1 above)
Comment (glyons-intel) [reply sycl/test/basic_tests/query.cpp]: OK.

**Review by smaslov-intel — APPROVED** [review]

**Pull request merged**